platform/posix/transport/src/plaintext_posix.c

```diff
@@ -113,30 +69,68 @@ int32_t Plaintext_Recv( const NetworkContext_t * pNetworkContext,
                         void * pBuffer,
                         size_t bytesToRecv )
 {
-    int32_t bytesReceived = 0;
+    int32_t bytesReceived = -1, selectStatus = -1, getTimeoutStatus = -1;
+    struct timeval transportTimeout;
+    socklen_t transportTimeoutLen;
+    fd_set readfds;
+
+    assert( pNetworkContext != NULL );
+    assert( pBuffer != NULL );
+    assert( bytesToRecv > 0 );
+
+    transportTimeoutLen = ( socklen_t ) sizeof( transportTimeout );
+    getTimeoutStatus = getsockopt( pNetworkContext->socketDescriptor,
```

Comment:
Can you add a comment about what this call is doing? Is it obtaining the receive timeout value?